### PR TITLE
Change port classification bit to ip address

### PR DESCRIPTION
**Pull request by noamblitz**
### Changes
Move port classification bit which allows for aggregate findings when the correct config ooi exists. 

Closes: https://github.com/minvws/nl-kat-coordination/issues/862

This pr also introduces a fix for ooi add form when field type is Dict[str,str] which is the case for Config.config.

Still needs an update of the rerun bits endpoint and test fixes.

### CODE REVIEW

**Comment by Lisser** [comment]
The recalculate bit logic, now also makes sure the Origins and OriginParameters exist in the graph.

This is done by:
- For each BitDefinition:
  - fetching all OOI's that are in the Bit's `consumes` field
  - For each OOI:
    - upsert Origin
    - For each Bit param:
      - query related objects according to the Path
        - For each related OOI
          - upsert OriginParameter

This is quite expensive..

**Comment by noamblitz** [comment]
Tested functionality and works fine. Will work on tests

File: octopoes/octopoes/core/service.py [hunk 1]
```diff
@@ -516,7 +523,42 @@ def get_scan_profile_inheritance(
...
+                for param_def in bit_definition.parameters:
+                    path = Path.parse(f"{param_def.ooi_type.get_object_type()}.{param_def.relation_path}").reverse()
+
+                    param_oois = self.ooi_repository.list_related(ooi, path, valid_time=valid_time)
+                    for param_ooi in param_oois:
+                        param = OriginParameter(origin_id=bit_instance.id, reference=param_ooi.reference)
+                        self.origin_parameter_repository.save(param, valid_time)
+
+        # remove all Origins and Origin Parameters, which are no longer in use
+
```
Comment (dekkers) [inline octopoes/octopoes/core/service.py:557]: Shouldn't there be code here that does this?

Comment (noamblitz) [reply octopoes/octopoes/core/service.py:557]: I think this is made to be only additive for now (all tough deleted bits will have no code and therefore return an empty list of OOIs which would me in work in practice).

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Adding the below Config OOI related to Network|internet, and then re-running all bits results in aggregated findings in both the web UI as well as the pdf. 
image
image
image

How Config OOI's can be added is currently undocumented and we should do this soon. Would be nice if this PR came with updated docs?

**Comment by underdarknl** [comment]
The Config OOI could be added via a Procedure.
To do this: We'd need a Bit that finds 'networks' without this specific Config available.
Then, this bit adds a QuestionOOI asking for the setting via a jsonschema defined form which in turn would result in a ConfigOOI on the related Network.
@Donnype this seems like a good initial usecase.

**Comment by noamblitz** [comment]
Agree!! Question: on ports found on this network, do you want a finding per port or one finding for all open ports. Altough a bit abstract, a perfect usecase for questions.

**Review by dekkers — APPROVED** [review]

**Pull request merged**